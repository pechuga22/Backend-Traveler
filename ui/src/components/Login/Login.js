import React from 'react'
import { Link } from 'react-router-dom'
import {
  Header,
  Button,
  Dimmer,
  Image,
  Container,
  Segment,
  Icon,
} from 'semantic-ui-react'

import traveler from '../../static/logo-traveler.svg'
import './Login.less'

export default function Login({ login }) {
  return (
    <Container text className='login'>
      <Segment basic>
        <Link to='/'>
          <Icon size='large' className='arrow left' color='brown' />
        </Link>
      </Segment>

      <Dimmer active inverted>
        <Segment basic padded>
          <Header as='h1' color='brown'>
            Welcome to
          </Header>
        </Segment>

        <Image src={traveler} size='medium' />
        <Segment basic padded='very'>
          <Header as='h2'>your experience start here</Header>
        </Segment>
        <Button color='pink' onClick={login} data-testid='login' circular>
          <h1>Login</h1>
        </Button>
      </Dimmer>
    </Container>
  )
}
