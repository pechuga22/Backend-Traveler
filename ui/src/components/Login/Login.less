@import '../../variables';

.ui.text.container.login {
  .ui.basic.segment a {
    position: relative;
    z-index: 2;

    &:hover {
      color: lighten(@greySecondary, 8%);
    }
  }

  .ui.inverted.dimmer {
    z-index: 1;
  }
  div.content {
    h2.header {
      margin-top: 20vh;
    }
  }
}
