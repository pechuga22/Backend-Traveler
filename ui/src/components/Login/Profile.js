import React from 'react'
import { Link } from 'react-router-dom'
import { Button, Segment, Dimmer, Comment, Header } from 'semantic-ui-react'
import Avatar from '../../static/avatar-traveler.png'
import './Profile.less'
export default function Profile({ logout, user }) {
  return (
    <Dimmer inverted active className='profile'>
      <Comment.Group size='massive'>
        <Comment>
          <Comment.Avatar src={Avatar} alt={user.profile.name} />
          <Comment.Author>
            <Header as='h1' color='blue'>
              Hey, {user.profile.given_name}
            </Header>
          </Comment.Author>
        </Comment>
      </Comment.Group>
      <Segment basic>
        <Header as='h1' color='brown'>
          Where would you like to go today?
        </Header>
        <Link to='/'>
          <Button color='blue' to='/' circular>
            <h1>Tell me</h1>
          </Button>
        </Link>
      </Segment>
      <Button color='pink' onClick={logout} circular>
        <h2>Logout</h2>
      </Button>
    </Dimmer>
  )
}
