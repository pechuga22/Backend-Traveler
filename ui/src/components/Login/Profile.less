@import '../../variables';

.ui.inverted.dimmer.profile {
  div.ui.massive.comments {
    margin: 0;
    .comment .author h1 {
      padding: 1rem;
      font-size: 1.6rem;
    }
    .comment .avatar {
      margin: 0;
    }
  }
  .ui.basic.segment.header {
    color: @greySecondary;
    font-weight: lighter;
  }
  a .ui.button {
    margin-top: 5vh;
    margin-bottom: 2rem;
  }
  .button {
    margin-top: 20vh;
  }
}
