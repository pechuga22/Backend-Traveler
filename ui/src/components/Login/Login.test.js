import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Login from './Login'

describe('./components/Login/Login.js', () => {
  describe('.render()', () => {
    const mockLogin = jest.fn()
    const component = shallow(<Login login={mockLogin} />)

    it('renders and matches our snapshot', () => {
      const component = shallow(<Login />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('find the login button', () => {
      const wrapper = component.find("[data-testid='login']")
      expect(wrapper.length).toBe(1)
    })

    it('click over the login button', () => {
      component.find('Button').simulate('click')
      expect(mockLogin).toBeCalled()
    })
  })
})
