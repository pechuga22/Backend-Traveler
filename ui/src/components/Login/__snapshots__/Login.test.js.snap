// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./components/Login/Login.js .render() renders and matches our snapshot 1`] = `
<Container
  className="login"
  text={true}
>
  <Segment
    basic={true}
  >
    <Link
      to="/"
    >
      <Icon
        as="i"
        className="arrow left"
        color="brown"
        size="large"
      />
    </Link>
  </Segment>
  <Dimmer
    active={true}
    inverted={true}
  >
    <Segment
      basic={true}
      padded={true}
    >
      <Header
        as="h1"
        color="brown"
      >
        Welcome to
      </Header>
    </Segment>
    <Image
      as="img"
      size="medium"
      src="logo-traveler.svg"
      ui={true}
    />
    <Segment
      basic={true}
      padded="very"
    >
      <Header
        as="h2"
      >
        your experience start here
      </Header>
    </Segment>
    <Button
      as="button"
      circular={true}
      color="pink"
      data-testid="login"
    >
      <h1>
        Login
      </h1>
    </Button>
  </Dimmer>
</Container>
`;
